pkg/asset/store.go

```diff
@@ -87,33 +118,43 @@ func (s *StoreImpl) load(dir string) error {
 	return nil
 }
 
-// LoadAssetFromState renders the asset object arguments from the state file contents.
-func (s *StoreImpl) LoadAssetFromState(asset Asset) error {
+// loadAssetFromState renders the asset object arguments from the state file contents.
+func (s *StoreImpl) loadAssetFromState(asset Asset) error {
 	bytes, ok := s.stateFileAssets[reflect.TypeOf(asset).String()]
 	if !ok {
 		return errors.Errorf("asset %q is not found in the state file", asset.Name())
 	}
 	return json.Unmarshal(bytes, asset)
 }
 
-// IsAssetInState tests whether the asset is in the state file.
-func (s *StoreImpl) IsAssetInState(asset Asset) bool {
+func (s *StoreImpl) removeAssetFromState(asset Asset) {
```

Comment:
This method is only used once (in `StoreImpl.Destroy`).  Maybe drop the method and just inline the `delete(...)` call?